proxy/src/main/java/com/wavefront/agent/queueing/TaskQueueFactoryImpl.java

```diff
@@ -110,15 +113,36 @@ public Long value() {
     try {
       File lockFile = new File(lockFileName);
       if (lockFile.exists()) {
-        Preconditions.checkArgument(true, lockFile.delete());
+        if (!Files.deleteIfExists(lockFile.toPath())) {
+          // no file with name found
+          throw new LogFilePathException();
+        }
       }
       FileChannel channel = new RandomAccessFile(lockFile, "rw").getChannel();
-      // fail if tryLock() returns null (lock couldn't be acquired)
-      Preconditions.checkNotNull(channel.tryLock());
-    } catch (Exception e) {
-      logger.severe("WF-005: Error requesting exclusive access to the buffer " +
+      if (channel.tryLock() == null) {
+        throw new OverlappingFileLockException();
+      }
+    } catch (LogFilePathException e){
+          logger.severe("Error finding buffer lock file " + lockFileName +
+          " - please make sure file path is correct and restart the proxy: " + e);
+      return new TaskQueueStub<>();
+    } catch (DirectoryNotEmptyException e) {
```

Comment:
Deleting catch as path will never be Directory (we create lockFile as new File above)